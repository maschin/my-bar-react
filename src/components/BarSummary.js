import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

export const BarSummary = ({ bottlesCount }) => {
    const bottlesWord = bottlesCount === 1 ? 'bottle' : 'bottles';

    return(
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">
                    You have total <span>{bottlesCount}</span> {bottlesWord} in your bar.
                </h1>
                <div className="page-header__actions">
                    <Link className="button" to="/create">Add Drink</Link>
                </div>
            </div>
        </div>
    );
};

const mapStateToProps = (state) => {
    return {
        drinks: state.drinks,
        bottlesCount: state.drinks.length
    };
};

export default connect(mapStateToProps)(BarSummary);