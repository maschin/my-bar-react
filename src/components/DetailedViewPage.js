import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import {
    openDrink,
    emptyDrink
} from '../actions/drinks';

export class DetailedView extends React.Component {
    onOpen = () => {
        this.props.openDrink(this.props.drink);
    }
    onMakeEmpty = () => {
        this.props.emptyDrink(this.props.drink);
    }
    render(){
        return(
            <div>
                {
                    this.props.drink ? (
                        <div>
                            <div className="page-header">
                                <div className="content-container">
                                    <div className="page-header__actions">
                                        <Link className="button" to={`/edit/${this.props.drink.id}`}>Edit</Link>
                                        { !this.props.drink.isOpen ? (
                                            <button id="openDrink" className="button" onClick={this.onOpen}>Open</button>
                                        ) : (

                                            !this.props.drink.isEmpty && <button id="emptyDrink" className="button" onClick={this.onMakeEmpty}>Empty</button>

                                        )}
                                    </div>
                                </div>
                            </div>
                            <div className="content-container">
                                <h1>{ this.props.drink.title }</h1>
                                <ul>
                                    <li>{ this.props.drink.category }</li>
                                    <li>{ this.props.drink.vol } L</li>
                                    <li>{ this.props.drink.alc } %</li>
                                    <li>{ this.props.drink.price }</li>
                                </ul>
                            </div>
                        </div>
                    ) : (
                        <h1 className="page-header__title">Record with <span>id:{this.props.match.params.id}</span> not found!</h1>
                )}
            </div>
        )
    }
}

const mapStateToProps = (state, props) => {
    return {
        drink: state.drinks.find((drink) => (drink.id === props.match.params.id))
    }
}

const mapDispatchToProps = (dispatch) => ({
        openDrink: (drink) => dispatch(openDrink(drink)),
        emptyDrink: (drink) => dispatch(emptyDrink(drink))
})

export default connect(mapStateToProps, mapDispatchToProps)(DetailedView);