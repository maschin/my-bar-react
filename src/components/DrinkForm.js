import React from 'react';

export class DrinkForm extends React.Component {
    onSubmit = () => {
        this.props.onSubmit();
    };
    render(){
        return (
          <form className="form" onSubmit={this.onSubmit}>
              <div>
                  <button className="button">{this.props.actionTitle}</button>
              </div>
          </form>
        );
    }
}

export default DrinkForm;