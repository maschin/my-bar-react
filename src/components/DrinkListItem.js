import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

export const DrinkListItem = ({
                                  id, title,
                                  category, vol,
                                  alc, price, aged,
                                  isOpen, isEmpty,
                                  boughtAt, bottledAt, bestBefore,
                              }) => (
    <div>
        <Link className={`list-item ${isOpen ? 'list-item--opened' : ''} ${isEmpty ? 'list-item--empty' : ''}`} to={`/view/${id}`}>
            <div>
                {title}
                <br/>
                {category} - {vol}L
            </div>
            <ul>
                <li>{alc}%</li>
                <li>{numeral(price/100).format('$0,0.00')}</li>
                <li>Age: {aged}</li>
            </ul>
            <ul>
                <li>Bought: {moment(boughtAt).format('MMMM Do, YYYY')}</li>
                <li>Bottled: {moment(bottledAt).format('MMMM Do, YYYY')}</li>
                <li>BBF: {moment(bestBefore).format('MMMM Do, YYYY')}</li>
            </ul>
        </Link>
    </div>
);



export default DrinkListItem;