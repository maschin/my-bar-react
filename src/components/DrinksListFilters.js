import React from 'react';
import { connect } from 'react-redux';
import {
    setTextFilter,
    setCategoryFilter
} from '../actions/drinksFilters';


export class DrinksListFilters extends React.Component {
    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value);
    }
    onCategoryChange = (e) => {
        this.props.setCategoryFilter(e.target.value);
    }
    render() {
        return (
            <div className="content-container">
                <div className="input-group">
                    <div className="input-group__item">
                        <input
                            type="text"
                            className="text-input"
                            placeholder="Search drinks"
                            value={this.props.drinksFilters.text}
                            onChange={this.onTextChange}
                        />
                    </div>
                    <div className="input-group__item">
                        <select
                            className="select"
                            value={this.props.drinksFilters.category}
                            onChange={e => this.onCategoryChange(e, 'category')}
                        >
                            <option value=''>-- All --</option>
                            {this.props.categories.map((category) =>
                                <option
                                    key={category}
                                    value={category}>
                                        {category}
                                </option>)}
                        </select>
                    </div>
                </div>
            </div>
        )
    }
}

DrinksListFilters.defaultProps = {
    categories: [
        'Absinthe',
        'Beer',
        'Brandy',
        'Cider',
        'Gin',
        'Horilka',
        'Liqueurs',
        'Mastika',
        'Ouzo',
        'Rakia',
        'Slivovitz',
        'Rum',
        'Sparkling Wine',
        'Tequila',
        'Vodka',
        'Vermouth',
        'Wine',
        'Whisky',
    ]
}

const mapStateToProps = (state) => {
    return {
        drinksFilters: state.drinksFilters
    }
}

const mapDispatchToProps = (dispatch) => ({
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    setCategoryFilter: (value) => dispatch(setCategoryFilter(value))
});

export default connect(mapStateToProps, mapDispatchToProps)(DrinksListFilters);