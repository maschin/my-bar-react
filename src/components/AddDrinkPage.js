import React from 'react';
import { connect } from 'react-redux';

import DrinkForm from '../components/DrinkForm';
import { startAddDrink } from '../actions/drinks';

export class AddDrinkPage extends React.Component {
    onSubmit = (drink) => {
        this.props.startAddDrink(drink);
        this.props.history.push('/');
    };
    render(){
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Add Bottle</h1>
                    </div>
                </div>
                <div className="content-container">
                    <DrinkForm
                        actionTitle='Add Bottle'
                        onSubmit={this.onSubmit}
                    />
                </div>
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => ({
    startAddDrink: (drink) => dispatch(startAddDrink(drink))
});

export default connect(undefined, mapDispatchToProps)(AddDrinkPage);