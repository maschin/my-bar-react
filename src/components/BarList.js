import React from 'react';
import { connect } from 'react-redux';

import DrinkListItem from '../components/DrinkListItem';

export const BarList = (props) => (
    <div className="content-container">
        <div className="list-header">
            <div className="show-for-desktop">Drink</div>
            <div className="show-for-desktop">Data</div>
        </div>
        <div className="list-body">
            {
                props.drinks.length === 0 ? (
                    <div className="list-item list-item--message">
                        <span>Your bar is totally empty!</span>
                    </div>
                ) : (
                    props.drinks.map((drink) => (
                       <DrinkListItem key={drink.id} {...drink} />
                    ))
                )
            }
        </div>
    </div>
);

const mapStateToProps = (state) => {
    return {
        drinks: state.drinks
    };
};

export default connect(mapStateToProps)(BarList);