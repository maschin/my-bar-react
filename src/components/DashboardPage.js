import React from 'react';
import BarList from '../components/BarList';
import BarSummary from '../components/BarSummary';

export const DashboardPage = () => (
    <div>
        <BarSummary />
        <BarList />
    </div>
);

export default DashboardPage;