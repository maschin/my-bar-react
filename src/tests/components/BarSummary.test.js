import React from 'react';
import { shallow } from 'enzyme';
import { BarSummary } from '../../components/BarSummary';

let wrapper;

test('should render BarSummary with one record correctly', () => {
    wrapper = shallow(<BarSummary bottlesCount={1} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render BarSummary with multiple records correctly', () => {
   wrapper = shallow(<BarSummary bottlesCount={123} />);
   expect(wrapper).toMatchSnapshot();
});

