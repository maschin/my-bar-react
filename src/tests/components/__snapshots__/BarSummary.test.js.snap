// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render BarSummary with multiple records correctly 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      You have total 
      <span>
        123
      </span>
       
      bottles
       in your bar.
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Drink
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render BarSummary with one record correctly 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      You have total 
      <span>
        1
      </span>
       
      bottle
       in your bar.
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Drink
      </Link>
    </div>
  </div>
</div>
`;
